doxia-core/src/main/java/org/apache/maven/doxia/parser/AbstractXmlParser.java

```diff
@@ -100,7 +98,7 @@
     private boolean validate = false;
 
     /** {@inheritDoc} */
-    public void parse( Reader source, Sink sink )
+    public void parse( Reader source, Sink sink, String reference )
```

Comment:
Does this change the public API? I'm not sure these changes are compatible.

Reply:
Please, refer to the JIRA for the conversation https://issues.apache.org/jira/browse/DOXIA-614

Reply:
The reference argument doesn't seem to be used here.

Reply:
Effectively, it is not. As explained in the JIRA issue the PR aims to be that the default method instead of the other one with 2 arguments. The way it is implemented right now, even when `Parser` interface has two method, the one with thre is not usable by how it's overriden in `AbstractParser` and never invoked in `DefaultDoxia`.
The final goal is to deprecate the one with two arguments so that any plugin can obtain the metadata of the sourcefile bien processed.

I am available to discuss online if anyone needs to, because I am bit confused by how this review is proceding. I think the JIRA issue, explanation here and the additional diagram provide a clear description of the goals and changes done which are the complex topics. However, I am seeing a focus on minor things. Furthermore, I raised 2 important points in the JIRA issue which had had no answer and which imo are worth discussing, one related to problems with the build and the other with testing.